### ISSUE TITLE
Discussion: Realtime sync in videos

### ISSUE THREAD

**Issue by iago1460** [kershner/screenBloom#13]
Hello, I think it will be possible to achive a perfect sync while playing videos, my idea is using the Multi monitor support + [two video instances](http://superuser.com/questions/1123192/how-to-play-two-instances-of-the-same-video-in-sync-with-a-custom-delay). 

Having two screens with the same video but one of them with a slightly delay (this is the video that will be displayed to the user) screenBloom will have enough time to proccess and send the data.

**Comment by kershner** [comment]
Hey there, pretty interesting idea.  So I guess you can do this right now if you have two monitors, right?  You just get two VLC video instances going, display one instance **muted** and fast-forwarded ~800ms on one monitor and set ScreenBloom to parse that display.  Then you put the actual un-muted video instance you're going to watch on the main display and hit play on both VLC instances.

I'll give that a try later.  The slight update delay comes from the network->Hue Bridge->Hue Lights interface, which means it might not be possible to set a perfectly consistently "delay" value in the dummy VLC instance.  It'll probably vary over the course of the movie.

**Comment by iago1460** [comment]
Yes, basically it is as you said, I am still working around how to get two video instances in sync with a custom delay between them

**Comment by khromov** [comment]
Had the same idea today and lo-and-behold, here's a thread about it. 😄  @iago1460 

I envision this functioning through an off-screen video decoder that decodes a video file at very low resolution (for example 320x200px), extracts the color data and sends it to Hue. It would then interface with real players (VLC, Plex) through an API. The real player would tell them where they are at time-wise and the off-screen decoder would play 100ms in advance of that file and send appropriate commands to the Hue.

Although the lag jitters a bit it, I still find it pretty stable so this has the potential to be a huge improvement over the current delay.
